src/libs/Navigation/AppNavigator/AuthScreens.tsx

```diff
@@ -224,7 +224,8 @@ function AuthScreens({session, lastOpenedPublicRoomID, initialLastUpdateIDApplie
         }
 
         if (!initialReportID) {
-            initialReportID = ReportUtils.findLastAccessedReport(!canUseDefaultRooms, shouldOpenOnAdminRoom(), activeWorkspaceID)?.reportID;
+            const initialReport = ReportUtils.findLastAccessedReport(!canUseDefaultRooms, shouldOpenOnAdminRoom(), activeWorkspaceID);
+            initialReportID = initialReport?.reportID ?? "";
```

Comment:
When a user logs in, initially there is no data in ONYX when 2FA is disabled with Xero (Because server not responding with any data when 2FA is required), in this case `initialReportID` will stay `undefined` and below code 
https://github.com/Expensify/App/blob/1908bd183b8d1953d81b69ce074c172f30826f90/src/libs/Navigation/AppNavigator/AuthScreens.tsx#L493
will result into `initialParams` as `r/undefined` and app will treat `undefined` as `reportId` which will result into `Hmm...not found page` in report. To prevent this scenario, we need to avoid keeping `initialReportID` as `undefined`. 
Hence above code will keep `initialReportID ` as `""` empty string when `initialReportID` is `undefined`.

This scenario is thoroughly explained in [proposal's](https://github.com/Expensify/App/issues/46026#issuecomment-2253555532) result section with video.